### PR TITLE
[webview_flutter] Support Flutter `TextInput`s

### PR DESCRIPTION
**Pull request by mklim**
## Description

Adds logic to `InputAwareWebView#clearFocus` to reset input creation
back to the container view's thread whenever the WebView is unfocused.
Fixes issue where Flutter UIs could no longer receive InputConnections
because WebView was causing all connections to happen on WebView's IME
thread.

Manually tested using mklim/plugins@928c790.

## Related Issues

flutter/flutter#19718

## Checklist

Before you create this PR confirm that it meets all requirements listed below by checking the relevant checkboxes (`[x]`). This will ensure a smooth and quick review process.

- [x] I read the [Contributor Guide] and followed the process outlined there for submitting PRs.
- [ ] My PR includes unit or integration tests for *all* changed/updated/fixed behaviors (See [Contributor Guide]).
- [x] All existing and new tests are passing.
- [x] I updated/added relevant documentation (doc comments with `///`).
- [x] The analyzer (`flutter analyze`) does not report any problems on my PR.
- [x] I read and followed the [Flutter Style Guide].
- [x] The title of the PR starts with the name of the plugin surrounded by square brackets, e.g. [shared_preferences]
- [x] I updated pubspec.yaml with an appropriate new version according to the [pub versioning philosophy].
- [x] I updated CHANGELOG.md to add a description of the change.
- [x] I signed the [CLA].
- [x] I am willing to follow-up on review comments in a timely manner.

## Breaking Change

Does your PR require plugin users to manually update their apps to accommodate your change?

- [ ] Yes, this is a breaking change (please indicate a breaking change in CHANGELOG.md and increment major revision).
- [x] No, this is *not* a breaking change.

[issue database]: https://github.com/flutter/flutter/issues
[Contributor Guide]: https://github.com/flutter/plugins/blob/master/CONTRIBUTING.md
[Flutter Style Guide]: https://github.com/flutter/flutter/wiki/Style-guide-for-Flutter-repo
[pub versioning philosophy]: https://www.dartlang.org/tools/pub/versioning
[CLA]: https://cla.developers.google.com/

### CODE REVIEW

File: packages/webview_flutter/android/src/main/java/io/flutter/plugins/webviewflutter/InputAwareWebView.java [hunk 1]
```diff
@@ -112,4 +112,28 @@ public void run() {
         });
     return super.checkInputConnectionProxy(view);
   }
+
+  /**
+   * Ensure that input creation happens back on {@link #containerView}'s thread once this view no
+   * longer has focus.
+   *
+   * <p>The logic in {@link #checkInputConnectionProxy} forces input creation to happen on Webview's
+   * thread for all connections. We undo it here so usres will be able to go back to typing in
```
Comment (amirh) [inline packages/webview_flutter/android/src/main/java/io/flutter/plugins/webviewflutter/InputAwareWebView.java]: nit typo: users

File: packages/webview_flutter/android/src/main/java/io/flutter/plugins/webviewflutter/InputAwareWebView.java [hunk 2]
```diff
@@ -112,4 +112,28 @@ public void run() {
...
+   */
+  @Override
+  public void clearFocus() {
+    super.clearFocus();
+    containerView.requestFocus();
+    containerView.post(
+        new Runnable() {
+          @Override
+          public void run() {
+            containerView.onWindowFocusChanged(true);
```
Comment (amirh) [inline packages/webview_flutter/android/src/main/java/io/flutter/plugins/webviewflutter/InputAwareWebView.java]: What do you think about only do this if we've set the ThreadedInputConnectionProxyAdapterView as the input target?

File: packages/webview_flutter/android/src/main/java/io/flutter/plugins/webviewflutter/InputAwareWebView.java (hunk 1 above)
Comment (mklim) [reply packages/webview_flutter/android/src/main/java/io/flutter/plugins/webviewflutter/InputAwareWebView.java]: Done.

File: packages/webview_flutter/android/src/main/java/io/flutter/plugins/webviewflutter/InputAwareWebView.java (hunk 2 above)
Comment (mklim) [reply packages/webview_flutter/android/src/main/java/io/flutter/plugins/webviewflutter/InputAwareWebView.java]: Makes sense. done.

File: packages/webview_flutter/android/src/main/java/io/flutter/plugins/webviewflutter/InputAwareWebView.java [hunk 3]
```diff
@@ -112,4 +112,32 @@ public void run() {
...
+    if (proxyAdapterView == null) {
+      // No need to reset the InputConnection to the default thread if we've never changed it.
+      return;
+    }
+    containerView.requestFocus();
+    containerView.post(
+        new Runnable() {
+          @Override
+          public void run() {
+            containerView.onWindowFocusChanged(true);
```
Comment (amirh) [inline packages/webview_flutter/android/src/main/java/io/flutter/plugins/webviewflutter/InputAwareWebView.java]: [nit, can definitely happen in a different PR]
We're repeating this magic sequence in more than one place, this specific call site isn't document so someone stepping into this code will probably be confused. Would probably be useful to extract these steps to a common method.

Comment (mklim) [reply packages/webview_flutter/android/src/main/java/io/flutter/plugins/webviewflutter/InputAwareWebView.java]: I'll clean it up in flutter/plugins#1853.

**Pull request merged**